src/cheri_insts.sail

```diff
@@ -950,29 +759,16 @@ union clause ast = CCopyType : (regidx, regidx, regidx)
  *   [CCSeal] knows how to work with these.
  */
 function clause execute (CCopyType(cd, cs1, cs2)) = {
-  let cs1_val = C(cs1);
+  let cs1_val' = C(cs1);
+  let cs1_val = invalidateCapUnless(cs1_val', not(isCapSealed(cs1_val')));
   let cs2_val = C(cs2);
-  let (cs1_base, cs1_top) = getCapBounds(cs1_val);
   let cs2_otype = unsigned(cs2_val.otype);
-  if not (cs1_val.tag) then {
-    handle_cheri_reg_exception(CapEx_TagViolation, cs1);
-    RETIRE_FAIL
-  } else if isCapSealed(cs1_val) then {
-    handle_cheri_reg_exception(CapEx_SealViolation, cs1);
-    RETIRE_FAIL
-  } else if hasReservedOType(cs2_val) then {
+  if hasReservedOType(cs2_val) then {
     C(cd) = int_to_cap(EXTS(cs2_val.otype));
     RETIRE_SUCCESS
-  } else if cs2_otype < cs1_base then {
```

Comment:
These checks have disappeared. I suppose they are subsumed by the less strict `invalidateCapUnless(newCap, representable)` below combined with the checks in `CCSeal`?

Reply:
Sounds sensible!